pkg/apis/kubeone/v1beta1/types.go

```diff
@@ -0,0 +1,331 @@
+	IsLeader          bool   `json:"isLeader"`
+
+	// If not provided (i.e. nil) defaults to TaintEffectNoSchedule, with key
+	// node-role.kubernetes.io/master for control plane nodes.
+	//
+	// Explicitly empty (i.e. []corev1.Taint{}) means no taints will be applied
+	// (this is default for worker nodes).
+	Taints []corev1.Taint `json:"taints,omitempty"`
+
+	// Information populated at the runtime
+	OperatingSystem OperatingSystemName `json:"-"`
+}
+
+// ControlPlaneConfig defines control plane nodes
+type ControlPlaneConfig struct {
+	Hosts []HostConfig `json:"hosts"`
+}
+
+// StaticWorkersConfig defines static worker nodes provisioned by KubeOne and kubeadm
+type StaticWorkersConfig struct {
+	Hosts []HostConfig `json:"hosts"`
+}
+
+// APIEndpoint is the endpoint used to communicate with the Kubernetes API
+type APIEndpoint struct {
+	// Host is the hostname on which API is running
+	Host string `json:"host"`
+
+	// Port is the port used to reach to the API
+	Port int `json:"port"`
+}
+
+// CloudProviderSpec describes the cloud provider that is running the machines.
+// Only one cloud provider must be defined at the single time.
+type CloudProviderSpec struct {
+	External     bool              `json:"external"`
+	CloudConfig  string            `json:"cloudConfig"`
+	AWS          *AWSSpec          `json:"aws"`
+	Azure        *AzureSpec        `json:"azure"`
+	DigitalOcean *DigitalOceanSpec `json:"digitalocean"`
+	GCE          *GCESpec          `json:"gce"`
+	Hetzner      *HetznerSpec      `json:"hetzner"`
+	Openstack    *OpenstackSpec    `json:"openstack"`
+	Packet       *PacketSpec       `json:"packet"`
+	Vsphere      *VsphereSpec      `json:"vsphere"`
+	None         *NoneSpec         `json:"none"`
+}
+
+// AWSSpec defines the AWS cloud provider
+type AWSSpec struct{}
+
+// AzureSpec defines the Azure cloud provider
+type AzureSpec struct{}
+
+// DigitalOceanSpec defines the DigitalOcean cloud provider
+type DigitalOceanSpec struct{}
+
+// GCESpec defines the GCE cloud provider
+type GCESpec struct{}
+
+// HetznerSpec defines the Hetzner cloud provider
+type HetznerSpec struct{}
+
+// OpenstackSpec defines the Openstack provider
+type OpenstackSpec struct{}
+
+// PacketSpec defines the Packet cloud provider
+type PacketSpec struct{}
+
+// VsphereSpec defines the vSphere provider
+type VsphereSpec struct{}
+
+// NoneSpec defines a none provider
+type NoneSpec struct{}
+
+// VersionConfig describes the versions of components that are installed on the machines
+type VersionConfig struct {
+	Kubernetes string `json:"kubernetes"`
+}
+
+// ClusterNetworkConfig describes the cluster network
+type ClusterNetworkConfig struct {
+	PodSubnet         string `json:"podSubnet"`
+	ServiceSubnet     string `json:"serviceSubnet"`
+	ServiceDomainName string `json:"serviceDomainName"`
+	NodePortRange     string `json:"nodePortRange"`
+	CNI               *CNI   `json:"cni,omitempty"`
+
+	// +optional
+	NetworkID string `json:"networkID,omitempty"`
```

Comment:
This needs to be dropped (or moved to HetznerSpec), as it's used in one single place, to assign hetzner CCM deployment an argument.

Reply:
I vote that we moved it to the `HetznerSpec` if it's Hetzner-specific.

Reply:
:lgtm: